extension/src/popup/components/account/AssetDetail/index.tsx

```diff
@@ -70,18 +73,40 @@ export const AssetDetail = ({
   subentryCount,
 }: AssetDetailProps) => {
   const dispatch: AppDispatch = useDispatch();
+  const [isNewAsset, setIsNewAsset] = useState(false);
   const isNative = selectedAsset === "native";
 
   const canonical = getAssetFromCanonical(selectedAsset);
   const isSorobanAsset = canonical.issuer && isSorobanIssuer(canonical.issuer);
-  const isOwnedScamAsset = useIsOwnedScamAsset(
-    canonical.code,
-    canonical.issuer,
-  );
 
   const { accountBalances: balances } = useSelector(
     transactionSubmissionSelector,
   );
+  const { scannedAsset } = useScanAsset(
```

Comment:
Does this need a loading state?
It seems like since the hook does an async thing to set `scannedAsset`, without the loading state won't `scannedAsset` only ever get assigned to the initial value in the hook?

Reply:
reviewed hook rules and see now that this value will get updated and cause a re-render, so never mind I think probably it does make sense to not represent the loading state.

Reply:
Yup, it should be triggering a re-render on value update